test/functional/ex_cmds/mksession_spec.lua

```diff
@@ -0,0 +1,50 @@
+-- Test generation of session files
+
+local helpers = require('test.functional.helpers')(after_each)
+
+
+describe('A session file', function()
+  -- The file name use for each session
+  local session_file = 'Xtest-functional-ex_cmds-mksession.vim'
+
+  before_each(function()
+    helpers.clear()
+  end)
+
+  -- Each test should start with a clean session file
+  after_each(function()
+    os.remove(session_file)
+  end)
+
+  it('restores tab-local working directories', function()
+    local tab_dir = 'Xtest-functional-ex_cmds-mksession_spec.tab'
+    local cwd_dir = helpers.call('getcwd')
+    lfs.mkdir(tab_dir)
```

Comment:
This should go to before_each, removing directory to after_each, just as any other cleanup.